src/Module.php

```diff
@@ -21,6 +21,6 @@ public function onBootstrap(MvcEvent $e)
         $events   = $app->getEventManager();
         $services = $app->getServiceManager();
 
-        $services->get(ContentValidationListener::class)->attach($events);
+        $services->get('ContentValidationListener')->attach($events);
```

Comment:
The aliases existed originally due to when ZF2 was originally developed and released. We tended to use just the class name at the time, minus the namespace; zend-servicemanager was also case insensitive. As we prepared for ZF3, we realized that these names were easy to mistype, and started moving to using `::class` notation, as static analyzers and IDEs will flag incorrectly typed class names when you use it, making them more safe.

The aliases we provide at this point are a legacy: they exist only to allow retrieval using the v2 naming conventions. The service itself **MUST** be named after the class, as that's the future direction.

What this means is that _in your own code_ you can override the entry such that the factory assigned will return an alternate implementation. If you do that, any of the aliases or the canonical name will return your variant.

As such, I am going to close this issue, as there is already a mechanism for supplying an alternative listener.

Reply:
Fair enough, thanks for having a look though :)